core/SymbolRef.h

```diff
@@ -29,19 +33,54 @@ class SymbolData : private DebugOnlyCheck<SymbolDataDebugCheck> {
 };
 CheckSize(SymbolData, 8, 8);
 
+class Symbol;
+
 class SymbolRef final {
     friend class GlobalState;
     friend class Symbol;
 
+    u4 _id;
+
 public:
-    SymbolRef(GlobalState const *from, u4 _id);
-    SymbolRef(const GlobalState &from, u4 _id);
+    // TODO: Enforce that this is only 3 bits?
+    enum class Kind : u1 {
+        ClassOrModule = 0,
+        Method = 1,
+        Field = 2,
+        TypeArgument = 3,
+        TypeMember = 4,
+    };
+
+    // Kind takes up this many bits in _id.
+    static constexpr u4 KIND_BITS = 3;
+    static constexpr u4 KIND_MASK = 0x7;
+
+    Kind kind() const {
+        return static_cast<Kind>(_id & KIND_MASK);
+    }
+
+    u4 id() const {
```

Comment:
This seems like a footgun to me now(as it looses information about type).
WDYT about calling it with a scarier name or marking it private altogather

Reply:
I made it private, and made a `*Index` accessor per symbol kind.